### PR TITLE
[T4PB-15011] replace/append partuuid

### PR DESCRIPTION
**Pull request by obi-t4**
# What is this PR?
This PR modifies the extlinux.conf operation:
- update only `PARTUUID={UUID}` part.
- when `PARTUUID={UUID}` part doesn't exist, it is just appended.

# issue
https://tier4.atlassian.net/browse/T4PB-15011

# test
rqx-58g and x86 platform have been confirmed.

### CODE REVIEW

File: app/ota_client.py [hunk 1]
```diff
@@ -724,7 +724,7 @@ def _create_regular_file(
         )
 
         if str(reginf.path).startswith("/boot"):
-            dst = boot_standby_path / reginf.path.name
+            dst = boot_standby_path / reginf.path.relative_to("/boot")
```
Comment (obi-t4) [inline app/ota_client.py:727]: to support directory under /boot.

File: app/extlinux_control.py [hunk 2]
```diff
@@ -474,45 +335,28 @@ def is_current_slot_marked_successful(self) -> bool:
...
+                        f"{ma_append.group(1)}APPEND{ma_rest.group(1)}root={self.standby_slot_partuuid}{ma_rest.group(3)}\n"
+                    )
+                else:
+                    # If root=PARTUUID= entry doesn't exist, it is just appended.
+                    out.append(f"{line} root={self.standby_slot_partuuid}\n")
+            else:
+                out.append(f"{line}\n")
+        with open(dst, "w") as f:
+            f.writelines(out)
+            os.fsync(f.fileno())
```
Comment (Bodong-Yang) [inline app/extlinux_control.py:359]: I think the following is enough:
```python
    def update_extlinux_cfg(self, dst: Path, src: Path):
       
        def _replace(ma: re.Match, repl: str):
            append_l = ma.group(0)
            res, n = re.compile(r'root=[\w\-=]*').subn(repl, append_l)
            if not n:
                res = f"{append_l} {repl}"

            return res

        _repl_func = partial(_replace, repl=self.standby_slot_partuuid)
        dst.write_text(
            re.compile(r'.*APPEND.*').sub(_repl_func, src.read_text())
        )
```

File: app/extlinux_control.py [hunk 3]
```diff
@@ -474,45 +336,19 @@ def is_current_slot_marked_successful(self) -> bool:
...
-        if src and src.is_file():
-            # load template extlinux_cfg from external
-            cfg_text = self.gen_extlinux_cfg(src)
-        else:
-            # generate extlinux from default settings
-            cfg_text = self.gen_extlinux_cfg()
-
-        with open(target, "w") as f:
-            f.write(cfg_text)
+        _repl_func = partial(_replace, repl=self.standby_slot_partuuid)
```
Comment (Bodong-Yang) [inline app/extlinux_control.py:350]:
```suggestion
        _repl_func = partial(_replace, repl=f"root=PARTUUID={self.standby_slot_partuuid}")
```
Sorry I forgot to prepare the pure PARTUUID into "root=PARTUUID=<PARTUUID>" in my previous comment.

File: app/extlinux_control.py [hunk 4]
```diff
@@ -474,45 +336,19 @@ def is_current_slot_marked_successful(self) -> bool:
...
+        dst.write_text(re.compile(r".*APPEND.*").sub(_repl_func, src.read_text()))
```
Comment (Bodong-Yang) [inline app/extlinux_control.py:351]:
```suggestion
        dst.write_text(re.compile(r'\n\s*APPEND.*').sub(_repl_func, src.read_text()))
```
I think of the regex again and I think this regex should runs faster than the previous one.

File: app/extlinux_control.py (hunk 3 above)
Comment (Bodong-Yang) [reply app/extlinux_control.py:350]: Yes I confirm that!

File: app/extlinux_control.py [hunk 5]
```diff
@@ -348,7 +348,7 @@ def _replace(ma: re.Match, repl: str):
             return res
 
         _repl_func = partial(_replace, repl=f"root={self.standby_slot_partuuid}")
-        dst.write_text(re.compile(r"\n\sAPPEND.*").sub(_repl_func, src.read_text()))
```
Comment (Bodong-Yang) [inline app/extlinux_control.py]: Emmm you just forgot to add * after \s, no need to change back to use `.*APPEND.*`.

**Comment by obi-t4** [comment]
@Bodong-Yang OTA on the actual board has been confirmed.

**Review by Bodong-Yang — APPROVED** [review]

**Pull request merged**